### PR TITLE
Read migration state from file in datalake

### PR DESCRIPTION
**Pull request by JesperGomes**
## References

* #0000

### CODE REVIEW

File: source/databricks/tests/entry_points/test_entry_points.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 
 def test__entry_point_uncommitted_migrations_count__returns_0(installed_package):
-    exit_code = subprocess.call(["uncommitted_migrations_count"])
+    exit_code = subprocess.call(["uncommitted_migrations_count", "-h"])
```
Comment (seblapela) [inline source/databricks/tests/entry_points/test_entry_points.py:19]: What is -h?

File: source/databricks/package/datamigration/uncommitted_migrations_count.py [hunk 2]
```diff
@@ -12,9 +12,129 @@
...
+
+    file_client = file_system_client.get_file_client(file_name)
+    download = file_client.download_file()
+    downloaded_bytes = download.readall()
+    return downloaded_bytes
+
+
+def _read_csv(bytes_data: bytes):
+    string_data = StringIO(bytes_data.decode())
+    return csv.reader(string_data, dialect="excel")
```
[the remark this answers is not in the document]
Comment (JesperGomes) [reply source/databricks/package/datamigration/uncommitted_migrations_count.py:79]: sure - I will look into that

File: source/databricks/package/datamigration/uncommitted_migrations_count.py [hunk 3]
```diff
@@ -12,9 +12,129 @@
...
+        args.data_storage_account_key,
+        args.wholesale_container_name,
+    )
+
     # This format is fixed as it is being used by external tools
     print(f"uncommitted_migrations_count={uncommitted_migrations_count}")
+
+
+if __name__ == "__main__":
+    start()
```
[the remark this answers is not in the document]
Comment (JesperGomes) [reply source/databricks/package/datamigration/uncommitted_migrations_count.py:140]: it's only needed for some of the unit tests that runs the py file directly

File: source/databricks/tests/entry_points/test_entry_points.py (hunk 1 above)
Comment (JesperGomes) [reply source/databricks/tests/entry_points/test_entry_points.py:19]: it calls 'help' and returns

File: source/databricks/package/datamigration/uncommitted_migrations_count.py [hunk 4]
```diff
@@ -12,9 +12,129 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from azure.identity import DefaultAzureCredential
+from azure.storage.filedatalake import DataLakeServiceClient, FileSystemClient
```
Comment (BjarkeMeier) [inline source/databricks/package/datamigration/uncommitted_migrations_count.py:16]: Are there some things here that need to be added to setup.py?

Comment (BjarkeMeier) [reply source/databricks/package/datamigration/uncommitted_migrations_count.py:16]: Can we add tests to catch such problems while running in the container?

File: source/databricks/package/datamigration/uncommitted_migrations_count.py [hunk 5]
```diff
@@ -12,9 +12,129 @@
...
+
+    committed_migrations = _download_committed_migrations(
+        storage_account_name, storage_account_key, container_name
+    )
+
+    all_migrations = _get_all_migrations()
+
+    uncommitted_migrations = [
+        m for m in all_migrations if m not in committed_migrations
+    ]
```
Comment (BjarkeMeier) [inline source/databricks/package/datamigration/uncommitted_migrations_count.py:118]: I think python allows something like this:
```suggestion
    uncommitted_migrations = all_migrations - committed_migrations
```

File: source/databricks/tests/entry_points/test_entry_points.py (hunk 1 above)
Comment (BjarkeMeier) [reply source/databricks/tests/entry_points/test_entry_points.py:19]: Nice 👍 So we can simplify the other places using `--only-validate-args`?

**Review by BjarkeMeier — APPROVED** [review]

File: source/databricks/package/datamigration/uncommitted_migrations_count.py (hunk 5 above)
Comment (JesperGomes) [reply source/databricks/package/datamigration/uncommitted_migrations_count.py:118]: ok - I prefer to leave it as is for now.

**Pull request merged**